### PR TITLE
Records in the job_modules table need to be unique

### PR DESCRIPTION
**Pull request by kraih**
This resolves the problem we were seeing on the staging machines with records in the`job_modules` table sometimes having duplicates if the worker->webui status update was slowed down due to heavy load (resulting in false negative job results).

I'm not really happy with the `try`/`catch` in `OpenQA::Schema::Result::Jobs`, but `DBIx::Class` does not currently support real PostgreSQL upserts, so it's probably the best we can do for now. I've tested the duplicate record cleanup SQL in `dbicdh/PostgreSQL/upgrade/80-81/001-unique_job_modules.sql` on OSD, and the migration there should take roughly 1-2 minutes (it is also safe to run the delete manually in advance).

### CODE REVIEW

**Comment by kraih** [comment]
This is the first migration i'm adding, so some extra scrutiny would be appreciated. I did test it with custom packages on the staging machines too though.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -1055,12 +1055,19 @@ sub update_backend {
...
     my ($self, $tm) = @_;
-    my $r = $self->modules->find_or_new({name => $tm->{name}});
-    if (!$r->in_storage) {
-        $r->category($tm->{category});
-        $r->script($tm->{script});
+
+    my $r;
+    my $modules = $self->modules;
+    try {
+        $r = $modules->new({name => $tm->{name}, category => $tm->{category}, script => $tm->{script}});
```
Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: isn't new+insert = create?

**Comment by coolo** [comment]
the migration looks good, but you need to rebase to get circleci config

**Comment by kraih** [comment]
I rebased right before opening the PR.

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (kraih) [reply lib/OpenQA/Schema/Result/Jobs.pm]: It should be, but there are overloads in the `JobModules` class, so i figured it's safer to stay close to the old invocation.

Comment (kraih) [reply lib/OpenQA/Schema/Result/Jobs.pm]: Looked at the code quickly, and `create` is a `shift->new_result(shift)->insert` call, so probably ok with our overloads.

**Comment by coolo** [comment]
right, it would have caused an error on circleci if you missed it. So it just wasn't triggered.

**Comment by coolo** [comment]
there was some CI hickup, so merging with superpower

**Pull request merged**